app/src/main/java/com/lagradost/cloudstream3/extractors/Chillx.kt

```diff
@@ -24,34 +22,30 @@ class Watchx : Chillx() {
     override val mainUrl = "https://watchx.top"
 }
 
+
+@Suppress("NAME_SHADOWING")
 open class Chillx : ExtractorApi() {
     override val name = "Chillx"
     override val mainUrl = "https://chillx.top"
     override val requiresReferer = true
-    private var key: String? = null
 
     override suspend fun getUrl(
         url: String,
         referer: String?,
         subtitleCallback: (SubtitleFile) -> Unit,
         callback: (ExtractorLink) -> Unit
     ) {
-        val master = Regex("\\s*=\\s*'([^']+)").find(
+        val master = Regex("""JScript[\w+]?\s*=\s*'([^']+)""").find(
             app.get(
                 url,
-                referer = referer ?: "",
-                headers = mapOf(
-                    "Accept" to "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8",
-                    "Accept-Language" to "en-US,en;q=0.5",
-                )
+                referer = url,
             ).text
         )?.groupValues?.get(1)
-        val decrypt = cryptoAESHandler(master ?: return, getKey().toByteArray(), false)?.replace("\\", "") ?: throw ErrorLoadingException("failed to decrypt")
-
+        val key = app.get("https://raw.githubusercontent.com/rushi-chavan/multi-keys/keys/keys.json").parsedSafe<Keys>()?.key?.get(0) ?: throw ErrorLoadingException("Unable to get key")
```

Comment:
Make this cached, we dont want to ddos github for every url.

Reply:
I have assigned that as key can be changed and previously as well its fetching from a Github json hosted on github so do you want me to assign key static in Extractor ?

Reply:
@phisher98 What I mean is that you should do something like this, as then you dont have to do an extra get request for every time you use this extractor
```kotlin
companion object {
  private val key : String? = null
  suspend fun getKey() {
    key ?: app.get("...").parsedSafe<Keys>()?.key?.get(0)?.also { key = it }
  }
}

```